src/gluonts/torch/distributions/quantile.py

```diff
@@ -0,0 +1,84 @@
+# Copyright 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License").
+# You may not use this file except in compliance with the License.
+# A copy of the License is located at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# or in the "license" file accompanying this file. This file is distributed
+# on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+# express or implied. See the License for the specific language governing
+# permissions and limitations under the License.
+
+from typing import Dict, List, Optional, Union
+
+import torch
+from torch.distributions import Distribution, constraints
+
+from gluonts.core.component import validated
+from gluonts.model.forecast_generator import (
+    ForecastGenerator,
+    QuantileForecastGenerator,
+)
+
+from .distribution_output import Output
+
+
+class QuantileDistribution(Distribution):
+    arg_constraints: Dict[str, constraints.Constraint] = {}
+
+    def __init__(
+        self,
+        outputs: torch.Tensor,
+        quantiles: Union[List[float], torch.Tensor],
+        validate_args=None,
+    ):
+        self.outputs = outputs
+        self.quantiles = torch.as_tensor(
+            quantiles, device=outputs.device, dtype=outputs.dtype
+        )
+        super().__init__(
+            batch_shape=outputs.shape[:-1], validate_args=validate_args
+        )
+
+    def quantile_loss(self, value: torch.Tensor) -> torch.Tensor:
+        value = value.unsqueeze(-1)
+        return 2 * (
+            (value - self.outputs)
+            * ((value <= self.outputs).float() - self.quantiles)
+        ).abs().sum(dim=-1)
+
+
+class QuantileOutput(Output):
```

Comment:
I am a bit unsure about the design here. The `QuantileOutput` object can, in principle, inherit from either
1. `DistributionOutput` or
2. `Output` 

## Justification for Option 1 (inherit from `DistributionOutput`)

Loss functions like [`torch.module.loss.QuantileLoss`](https://github.com/awslabs/gluonts/blob/dev/src/gluonts/torch/modules/loss.py#L85) expect a `torch.distributions.Distribution` object when computing loss. Therefore, we need to create a class `QuantileDistribution` (defined above) that holds the quantile predictions in it. 

This could imply that `QuantileOutput` should inherit from `DistributionOutput` since it produces a `QuantileDistribution` object.

## Justification for Option 2 (inherit from `Output`)

We can differentiate between `Output` and `DistributionOutput` objects as follows:

- `DistributionOutput` objects allow sampling. Therefore, they can be used in sample-based models like DeepAR as well as Seq2Seq models (e.g., TFT, PatchTST)
- `Output` may not necessarily allow sampling. Therefore, they are only suitable for Seq2Seq models (e.g., TFT, PatchTST)

I am now leaning towards Option 2

Reply:
I’m also leaning towards option 2, but this makes me think whether the `QuantileLoss` and `NegativeLogLikelihood` kind of classes are really useful: you can’t really mix and match them with different outputs, they are kind of tied

Reply:
I agree, it seems that most classes that inherit from `torch.distributions.Distribution` usually only implement one of `log_prob`, `quantile_loss`, `crps`, `energy_score`, so there is tight coupling between loss/output. In theory, one could implement multiple methods for the same distribution object - but that would require a lot of work.

Reply:
> A better alternative could be defining loss method as part of an Output object, and setting it to NLL for DistrbutionOutput.

Yes, that seems like a better approach. Then subclasses of `Output` define everything about the output projection and how it (and everything else) is learned.

As far as I can see, this looks much cleaner than how things are now (and worth the breaking change imo)

Reply:
Okay, then I will update this PR to combine the Output class with loss calculation.